features/misc.go

```diff
@@ -93,3 +96,32 @@ var haveV3ISA = internal.NewFeatureTest("v3 ISA", func() error {
 		},
 	})
 }, "5.1")
+
+// HaveV4ISA probes the running kernel if instructions of the v4 ISA are supported.
+//
+// Upstream commit 1f9a1ea821ff ("bpf: Support new sign-extension load insns").
+//
+// See the package documentation for the meaning of the error return value.
+func HaveV4ISA() error {
+	return haveV4ISA()
+}
+
+var haveV4ISA = internal.NewFeatureTest("v4 ISA", func() error {
+	err := probeProgram(&ebpf.ProgramSpec{
+		Type: ebpf.SocketFilter,
+		Instructions: asm.Instructions{
+			asm.Mov.Imm(asm.R0, 0),
+			asm.JEq.Imm(asm.R0, 1, "error"),
+			asm.LongJump("exit"),
+			asm.Mov.Imm(asm.R0, 1).WithSymbol("error"),
+			asm.Return().WithSymbol("exit"),
+		},
+	})
+	if errors.Is(err, sys.ENOTSUPP) {
```

Comment:
I'm on the fence about putting this here. There's a big chance our v2/v3 probes are currently broken on arm as well, but we lack the possibility to test them. Also the versions affected are ancient and probably not very well-supported on recent arm devices.

Let's add this as a potential retcode to probeProgram. It's specific enough to make sense and not trigger any false positives (it screams 'unsupported', after all 😉).

Reply:
Changed the error-catching logic. Unfortunately, had to add the case for `ebpf.StructOps` for consistency, since we expect it to return nil rather than `ebpf.ErrNotSupp` with that program type. Let me know!